src/FirebirdSql.Data.FirebirdClient/Types/FbZonedDateTime.cs

```diff
@@ -72,8 +72,50 @@ public override int GetHashCode()
 		}
 	}
 
-	public bool Equals(FbZonedDateTime other) => DateTime.Equals(other.DateTime) && TimeZone.Equals(other.TimeZone, StringComparison.OrdinalIgnoreCase);
-
+	public bool Equals(FbZonedDateTime other) => DateTime.Equals(other.DateTime) && TimeZone.Equals(other.TimeZone, StringComparison.OrdinalIgnoreCase);
+
+	TypeCode IConvertible.GetTypeCode() => TypeCode.Object;
+
+	bool IConvertible.ToBoolean(IFormatProvider provider) => throw new InvalidCastException(nameof(Boolean));
+
+	byte IConvertible.ToByte(IFormatProvider provider) => throw new InvalidCastException(nameof(Byte));
+
+	char IConvertible.ToChar(IFormatProvider provider) => throw new InvalidCastException(nameof(Char));
+
+	DateTime IConvertible.ToDateTime(IFormatProvider provider) => DateTime;

```

Comment:
Put all the "real" implementations together above all the "throw" implementations.